### PR TITLE
Feature/sup 52/item

### PR DESCRIPTION
**Pull request by FortuneAwaker**
Category and Item components for frontend.

### CODE REVIEW

**Review by IrynaZazyba — COMMENTED** [review]
Need to implement remove items

File: frontend/src/App.js [hunk 1]
```diff
@@ -29,11 +31,27 @@ function App() {
...
+            return <Category/>
+        })}/>
+    };
+
     return (
         <UserContext>
             <Header/>
             <Switch>
                 <Route exact path='/' component={Login}/>
+                <Route path={'/customers/' + currentCustomerId + '/category'} render={renderCategory}/>/>
```
Comment (eugenmolchanov) [inline frontend/src/App.js]: Please, use es6 string concatenation.

File: frontend/src/Header.js [hunk 2]
```diff
@@ -12,6 +12,8 @@ function Header() {
 
     const profileClass = window.location.pathname.match(/.profile/) ? "active" : "";
     const customersClass = window.location.pathname === "/customers" ? "active" : "";
+    const categoryClass = window.location.pathname.match(/.category/) ? "active" : "";
+    const itemClass = window.location.pathname.match(/.item/) ? "active" : "";
```
Comment (eugenmolchanov) [inline frontend/src/Header.js]: We can create a function with one param for this statement and replace all the above lines (window.location.pathname.match ...) with it.

File: frontend/src/Header.js [hunk 3]
```diff
@@ -24,6 +26,15 @@ function Header() {
                     </Nav.Link>}
                     {checkPermission && user.role === "ROLE_SYSTEM_ADMIN" &&
                     <Nav.Link className={customersClass} href="/customers">Customers</Nav.Link>}
+                    {checkPermission && (user.role === "ROLE_SYSTEM_ADMIN" || user.role === "ROLE_DISPATCHER" ||
```
Comment (eugenmolchanov) [inline frontend/src/Header.js]: checkPermission && (user.role === "ROLE_SYSTEM_ADMIN" || user.role === "ROLE_DISPATCHER" ||
                        user.role === "ROLE_LOGISTICS_SPECIALIST" || user.role === "ROLE_ADMIN" can be extracted in a separate variable

File: frontend/src/pages/category/Category.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+import Table from "react-bootstrap/Table";
+import Button from "react-bootstrap/Button";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import TogglePage from "../../components/TogglePage";
+import CardContainer from "../../components/CardContainer";
+
+export default () => {
+
+    const [currentCustomerId, setSelected] = useState(JSON.parse(localStorage.getItem('user')).customers[0].id);
```
Comment (eugenmolchanov) [inline frontend/src/pages/category/Category.js:18]: What if user is not in the local storage or user doesn't have customers?

File: frontend/src/pages/category/ModalAddCategory.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        }));
+        if (validationResult.length === 0) {
+            fetch('/customers/' + currentCustomerId + '/category', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(categoryDto)
+            })
+                .then(function (response) {
```
Comment (eugenmolchanov) [inline frontend/src/pages/category/ModalAddCategory.js]: We may use arrow function instead.

**Review by eugenmolchanov — CHANGES_REQUESTED** [review]
Please, see the comments. Also please look at feature/user-management-page #38. There are many common notes. Please, fix corresponding issues in this pr.

File: frontend/src/pages/category/Category.js (hunk 4 above)
Comment (FortuneAwaker) [reply frontend/src/pages/category/Category.js:18]: changed everywhere to 
    const currentCustomerId = localStorage.
    getItem("currentCustomerId") != null ? localStorage.
    getItem("currentCustomerId"): 0;

File: frontend/src/Header.js (hunk 2 above)
Comment (FortuneAwaker) [reply frontend/src/Header.js]: done

File: frontend/src/App.js (hunk 1 above)
Comment (FortuneAwaker) [reply frontend/src/App.js]: done

File: frontend/src/pages/category/Category.js (hunk 4 above)
Comment (FortuneAwaker) [reply frontend/src/pages/category/Category.js:18]: done

File: frontend/src/Header.js (hunk 3 above)
Comment (FortuneAwaker) [reply frontend/src/Header.js]: done

File: frontend/src/pages/category/ModalAddCategory.js (hunk 5 above)
Comment (FortuneAwaker) [reply frontend/src/pages/category/ModalAddCategory.js]: done

**Review by eugenmolchanov — APPROVED** [review]

**Review by IrynaZazyba — APPROVED** [review]
According to TR:  should be an opportunity to remove several items at once.

**Pull request merged**